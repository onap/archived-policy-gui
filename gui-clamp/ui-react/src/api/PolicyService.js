/*-
 * ============LICENSE_START=======================================================
 * ONAP POLICY-CLAMP
 * ================================================================================
 * Copyright (C) 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class PolicyService {
  static getPoliciesList() {
    return fetch(window.location.pathname + 'restservices/clds/v2/policies', {
        method: 'GET',
        credentials: 'same-origin'
        })
        .then(function(response) {
            console.debug("getPoliciesList response received: ", response.status);
            if (response.ok) {
                console.info("getPoliciesList query successful");
                return response.json();
            } else {
                return response.text().then(responseBody => {
                    throw new Error("HTTP " + response.status + "," + responseBody);
                })
            }
        })
        .catch(function(error) {
            console.error("getPoliciesList error occurred ", error);
            alert("getPoliciesList error occurred " + error);
            return undefined;
        })
  }
  static createNewPolicy(policyModelType, policyModelVersion, policyName, policyVersion, policyJson) {
    return fetch(window.location.pathname + 'restservices/clds/v2/policies/' + policyModelType + '/'
                    + policyModelVersion + '/' + policyName + '/' + policyVersion, {
            method: 'POST',
            credentials: 'same-origin',
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(policyJson)
        })
        .then(function (response) {
            console.debug("createNewPolicy response received: ", response.status);
            if (response.ok) {
                console.info("createNewPolicy query successful");
                return response.text();
            } else {
               return response.text().then(responseBody => {
                    throw new Error("HTTP " + response.status + "," + responseBody);
                })
            }
        })
        .catch(function (error) {
            console.error("createNewPolicy error occurred ", error);
            alert ("createNewPolicy error occurred " + error);
            return undefined;
        });
  }
  static deletePolicy(policyModelType, policyModelVersion, policyName, policyVersion) {
    return fetch(window.location.pathname + 'restservices/clds/v2/policies/' + policyModelType + '/'
        + policyModelVersion + '/' + policyName + '/' + policyVersion, {
            method: 'DELETE',
            credentials: 'same-origin'
        })
        .then(function (response) {
            console.debug("deletePolicy response received: ", response.status);
            if (response.ok) {
                console.info("deletePolicy query successful");
                return response.text();
            } else {
                return response.text().then(responseBody => {
                    throw new Error("HTTP " + response.status + "," + responseBody);
                })
            }
        })
        .catch(function (error) {
            console.error("deletePolicy error occurred ", error);
            alert ("deletePolicy error occurred " + error);
            return undefined;
        });
  }
}
