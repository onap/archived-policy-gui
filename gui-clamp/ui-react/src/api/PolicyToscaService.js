/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019, 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class PolicyToscaService {
  static getToscaPolicyModels() {
    return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', { method: 'GET', credentials: 'same-origin' })
      .then(function (response) {
        console.debug("getToscaPolicyModels response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getToscaPolicyModels query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getToscaPolicyModels error received", error);
        return {};
      });
  }

  static getToscaPolicyModelYaml(policyModelType, policyModelVersion) {
        return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/yaml/' + policyModelType + "/" + policyModelVersion, {
            method: 'GET',
            credentials: 'same-origin'
        })
            .then(function (response) {
                console.debug("getToscaPolicyModelYaml response received: ", response.status);
                if (response.ok) {
                    return response.json();
                } else {
                    console.error("getToscaPolicyModelYaml query failed");
                    return "";
                }
            })
            .catch(function (error) {
                console.error("getToscaPolicyModelYaml error received", error);
                return "";
            });
 }

 static getToscaPolicyModel(policyModelType, policyModelVersion) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + "/" + policyModelVersion, {
             method: 'GET',
             credentials: 'same-origin'
         })
             .then(function (response) {
                 console.debug("getToscaPolicyModel response received: ", response.status);
                 if (response.ok) {
                     return response.json();
                 } else {
                     console.error("getToscaPolicyModel query failed");
                     return {};
                 }
             })
             .catch(function (error) {
                 console.error("getToscaPolicyModel error received", error);
                 return {};
             });
  }

  static createPolicyModelFromToscaModel(jsonData) {
       return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', {
           method: 'POST',
           credentials: 'same-origin',
           headers: {
             "Content-Type": "a",
           },
           body: JSON.stringify(jsonData)
         })
         .then(function(response) {
           console.debug("createPolicyModelFromToscaModel response received: ", response.status);
           if (response.ok) {
             var message = {
               status: response.status,
               message: 'Tosca Policy Model successfully uploaded'
             };
             return message;
           } else {
             console.error("createPolicyModelFromToscaModel failed");
             return response.text();
           }
         })
         .catch(function(error) {
           console.error("createPolicyModelFromToscaModel error received", error);
           return "";
         });
     }

     static updatePolicyModelTosca(policyModelType, policyModelVersion, jsonData) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + '/' + policyModelVersion, {
             method: 'PUT',
             credentials: 'same-origin',
             headers: {
               "Content-Type": "a",
             },
             body: JSON.stringify(jsonData)
           })
           .then(function(response) {
             console.debug("updatePolicyModelTosca response received: ", response.status);
             if (response.ok) {
               var message = {
                 status: response.status,
                 message: 'Tosca Policy Model successfully uploaded'
               };
               return message;
             } else {
               console.error("updatePolicyModelTosca failed");
               return response.text();
             }
           })
           .catch(function(error) {
             console.error("updatePolicyModelTosca error received", error);
             return "";
           });
       }
}
