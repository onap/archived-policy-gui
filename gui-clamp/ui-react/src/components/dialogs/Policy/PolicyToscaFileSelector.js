/*-
 * ============LICENSE_START=======================================================
 * ONAP POLICY-CLAMP
 * ================================================================================
 * Copyright (C) 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

import React from 'react'
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import styled from 'styled-components';
import Alert from 'react-bootstrap/Alert';
import { Input, InputLabel, Button, SvgIcon } from "@material-ui/core";
import PublishIcon from '@material-ui/icons/Publish';
import PolicyService from '../../../api/PolicyService';

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const StyledMessagesDiv = styled.div`
  overflow: auto;
  max-height: 300px;
`


export default class PolicyToscaFileSelector extends React.Component {

  state = {
    show: this.props.show,
    alertMessages: [],
  }

  constructor(props, context) {
    super(props, context);
    this.handleClose = this.handleClose.bind(this);
    this.onFileChange = this.onFileChange.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (this.props.show !== this.state.show) {
      this.setState({ show: this.props.show });
    }
  }

  handleClose() {
    this.props.disableFunction();
    this.setState({ alertMessages: [] });
  }

  onFileChange(target) {
    this.setState({ alertMessages: [] });
    target.currentTarget.files.forEach(file => {
      const fileReader = new FileReader();
      fileReader.readAsDataURL(file);
      fileReader.onload = (content) => {
        PolicyService.sendNewPolicyModel(atob(content.target.result.split(",")[1])).then(respModelCreate => {
          if (typeof (respModelCreate) === "undefined") {
            //it indicates a failure
            this.setState(state => {
              return {
                alertMessages: [...state.alertMessages, (<Alert variant="danger"><Alert.Heading>{ file.name }</Alert.Heading><p>Policy Tosca Model Creation Failure</p>
                  <hr/>
                  <p>Type: { file.type }</p><p>Size: { file.size }</p></Alert>)]
              };
            });
          } else {
            this.props.toscaTableUpdateFunction();
            this.setState(state => {
              return {
                alertMessages: [...state.alertMessages, (<Alert variant="success"><Alert.Heading>{ file.name }</Alert.Heading><p>Policy Tosca Model Created Successfully</p>
                  <hr/>
                  <p>Type: { file.type }</p><p>Size: { file.size }</p></Alert>)]
              };
            });
          }
        });
      };
    });

  }

  render() {
    return (
      <ModalStyled size="lg" show={ this.state.show } onHide={ this.handleClose } backdrop="static" keyboard={ false }>
        <Modal.Header closeButton>
          <Modal.Title>Create New Policy Tosca Model</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form.Group as={ Row } controlId="formPlaintextEmail">
            <Col sm="10">
              <input type="file" multiple accept=".yaml,.yml" id="fileUploadButton" style={ { display: 'none' } } onChange={ this.onFileChange }/>
              <label htmlFor={ 'fileUploadButton' }>
                <Button color="primary" variant="contained" component="span"
                        startIcon={
                          <SvgIcon fontSize="small">
                            <PublishIcon/>
                          </SvgIcon>
                        }>
                  Upload Files
                </Button>
                <p>(Only YAML files are supported)</p>
              </label>
              <StyledMessagesDiv>
                { this.state.alertMessages }
              </StyledMessagesDiv>
            </Col>
          </Form.Group>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={ this.handleClose }>Close</Button>
        </Modal.Footer>
      </ModalStyled>
    );
  }
}
