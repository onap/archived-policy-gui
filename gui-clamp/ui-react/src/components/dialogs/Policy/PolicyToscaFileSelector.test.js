/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2022 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import PolicyToscaFileSelector from "./PolicyToscaFileSelector";
import PolicyService from "../../../api/PolicyService";
import Alert from "react-bootstrap/Alert";
import PolicyToscaService from "../../../api/PolicyToscaService";
import fs from "fs";

describe('Verify PolicyToscaFileSelector', () => {

    let toscaPoliciesList = fs.readFileSync('src/components/dialogs/Policy/toscaPoliciesList.test.json', {
        encoding: 'utf8',
        flag: 'r'
    });

    const uploadFile = JSON.stringify(
        {
            name: 'test-file.yaml',
            lastModified: 1639414348371,
            lastModifiedDate: 'Sat Jan 1 2022 00:00:01 GMT+0000',
            size: 32880,
            type: ''
        });

    const file = new Blob([uploadFile], {type: 'file'})
    const logSpy = jest.spyOn(console, 'log');

    const alertMessages = [(
        <Alert variant="success"><Alert.Heading>{file.name}</Alert.Heading><p>Policy Tosca Model Creation Test</p>
            <hr/>
            <p>Type: {file.type}</p><p>Size: {file.size}</p></Alert>)];

    it("renders correctly", () => {
        const component = shallow(<PolicyToscaFileSelector/>);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Test handleClose', async () => {
        const flushPromises = () => new Promise(setImmediate);
        const showFileSelectorMock = jest.fn();
        const showFileSelector = showFileSelectorMock.bind({
            showFileSelector: false
        });
        const handleClose = jest.spyOn(PolicyToscaFileSelector.prototype, 'handleClose');
        const component = shallow(<PolicyToscaFileSelector disableFunction={showFileSelector} show={false}/>);

        component.find('[variant="secondary"]').get(0).props.onClick();
        await flushPromises();
        component.update();

        expect(handleClose).toHaveBeenCalledTimes(1);
        expect(component.state('show')).toEqual(false);

        handleClose.mockClear();
    });

    it('handleClose called when top-right button clicked', async () => {
        const flushPromises = () => new Promise(setImmediate);
        const showFileSelectorMock = jest.fn();
        const showFileSelector = showFileSelectorMock.bind({
            showFileSelector: false
        });
        const handleClose = jest.spyOn(PolicyToscaFileSelector.prototype, 'handleClose');
        const component = shallow(<PolicyToscaFileSelector disableFunction={showFileSelector} show={false}/>);

        component.find('[size="lg"]').get(0).props.onHide();
        await flushPromises();
        component.update();

        expect(handleClose).toHaveBeenCalledTimes(1);
        expect(component.state('show')).toEqual(false);

        handleClose.mockClear();
    });

    it("onFileChange called when upload button clicked", () => {
        PolicyService.sendNewPolicyModel = jest.fn().mockImplementation(() => {
            return Promise.resolve({});
        });

        const component = shallow(<PolicyToscaFileSelector/>);
        const instance = component.instance();
        const target = {
            currentTarget: {files: [file]}
        }

        instance.onFileChange(target);

        component.find('[type="file"]').get(0).props.onChange(target);
        expect(logSpy).toHaveBeenCalledWith('onFileChange target');
    });

    it("setAlertMessage state", () => {
        PolicyToscaService.getToscaPolicyModels = jest.fn().mockImplementation(() => {
            return Promise.resolve(toscaPoliciesList);
        });
        const getAllToscaModelsMock = jest.fn();
        const getAllToscaModels = getAllToscaModelsMock.bind({
            toscaModelsListData: toscaPoliciesList,
            toscaModelsListDataFiltered: toscaPoliciesList
        });

        const component = shallow(<PolicyToscaFileSelector toscaTableUpdateFunction={getAllToscaModels}/>);
        component.setState({alertMessages: alertMessages});

        expect(component.state('alertMessages')).toEqual(alertMessages);
    });
});