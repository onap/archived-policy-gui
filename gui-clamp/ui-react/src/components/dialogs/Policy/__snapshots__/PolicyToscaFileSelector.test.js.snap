// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify PolicyToscaFileSelector renders correctly 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Create New Policy Tosca Model
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formPlaintextEmail"
    >
      <Col
        sm="10"
      >
        <input
          accept=".yaml,.yml"
          id="fileUploadButton"
          multiple={true}
          onChange={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          type="file"
        />
        <label
          htmlFor="fileUploadButton"
        >
          <WithStyles(ForwardRef(Button))
            color="primary"
            component="span"
            startIcon={
              <WithStyles(ForwardRef(SvgIcon))
                fontSize="small"
              >
                <Memo />
              </WithStyles(ForwardRef(SvgIcon))>
            }
            variant="contained"
          >
            Upload Files
          </WithStyles(ForwardRef(Button))>
          <p>
            (Only YAML files are supported)
          </p>
        </label>
        <styled.div />
      </Col>
    </FormGroup>
  </ModalBody>
  <ModalFooter>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="secondary"
    >
      Close
    </WithStyles(ForwardRef(Button))>
  </ModalFooter>
</Styled(Modal)>
`;
